pkg/kube/watcher.go

```diff
@@ -258,6 +267,10 @@ func isTimeout(err error) bool {
 	return false
 }
 
+func isAPINotReady(err error) bool {
+	return strings.Contains(err.Error(), "apiserver not ready")
```

Comment:
It would be good to have a comment about where this comes from (k3s, over HTTP so we can't test for a specific error object). Ideally we also check that `err` is of the right type here, but I guess that can be optional.